### PR TITLE
[KYUUBI #6615] Make Jetty sending server version in response configurable

### LINKED ISSUE #6615: [FEATURE] Kyuubi Jetty server should be able to hide version info in response, thus avoiding vulnerability exposure.
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

Add a switch for enabling/disabling jetty sending server version in response.

### Motivation

By default Jetty always send version in ressponse, such as Jetty(9.4.54.v20240208), which could expose vulnerablilty. Malicious visitors could easily detect Jetty version and then perform specific attack.

### Describe the solution

Add a config item that controls whether Jetty should send its version in response.

Sending Jetty version could be disabled by calling HttpConfiguration::setSendServerVersion(false)

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by paul8263**
# :mag: Description
## Issue References 🔗

This pull request fixes #6615

## Describe Your Solution 🔧

Add a config item that controls whether Jetty should send its version in response.

Sending Jetty version could be disabled by calling HttpConfiguration::setSendServerVersion(false)

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

Compiled and tested manually.

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/ui/JettyServer.scala [hunk 1]
```diff
@@ -78,7 +78,8 @@ object JettyServer {
       host: String,
       port: Int,
       poolSize: Int,
-      stopTimeout: Long): JettyServer = {
+      stopTimeout: Long,
+      shouldSendJettyVersion: Boolean = true): JettyServer = {
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/ui/JettyServer.scala:82]: shouldSendJettyVersion => sendServerVersion

do we really need such a default value?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -700,6 +700,14 @@ object KyuubiConf {
       .booleanConf
       .createWithDefault(true)
 
+  val FRONTEND_JETTY_SEND_VERSION_ENABLED: ConfigEntry[Boolean] =
+    buildConf("kyuubi.frontend.jetty.send.version.enabled")
+      .doc("When true, Kyuubi will send Jetty version in HTTP response, " +
+        "which may expose vulnerability")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:706]: > which may expose vulnerability

please avoid such thrilling words in user-facing docs

```suggestion
      .doc("Whether to send Jetty version in HTTP response.")
```

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 3]
```diff
@@ -700,6 +700,14 @@ object KyuubiConf {
       .booleanConf
       .createWithDefault(true)
 
+  val FRONTEND_JETTY_SEND_VERSION_ENABLED: ConfigEntry[Boolean] =
+    buildConf("kyuubi.frontend.jetty.send.version.enabled")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:704]:
```suggestion
    buildConf("kyuubi.frontend.jetty.sendVersion.enabled")
```

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/ui/JettyServer.scala (hunk 1 above)
Comment (paul8263) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/ui/JettyServer.scala]: Hi @pan3793 ，

Thanks for your suggestion. As long as there are only 2 places calling the constructor of JettyServer. We could remove the default value for the new introduced sendServerVersion parameter.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
@paul8263 please run `dev/gen/gen_all_config_docs.sh` to regenerate docs.

**Comment by paul8263** [comment]
> @paul8263 please run `dev/gen/gen_all_config_docs.sh` to regenerate docs.

Thanks. It should be good as there is a way to generate the doc. In the previous commit I editted the doc manually, but put the new configuration in the wrong chapter. The latest commit should be correct.

**Comment by pan3793** [comment]
Thanks, merged to master/1.9

**Pull request closed without merging**